tests/tests/test-txpool/test-txpool-fairness.ts

```diff
@@ -0,0 +1,273 @@
+// be used as-is, e.g. it must be at least (block.baseFee + maxPriorityFeePerGas)
+const HIGH_MAX_FEE_PER_GAS = "0x" + GLMR.toString(16);
+
+describeDevMoonbeam("Tip should be respected", (context) => {
+  it("should prefer txn with higher tip", async function () {
+    const NO_TIP = 0n;
+    const MED_TIP = 5n * MILLIGLMR;
+    const HIGH_TIP = 20n * MILLIGLMR;
+
+    await context.polkadotApi.tx.balances
+      .transfer(dorothy.address, GLMR)
+      .signAndSend(alith, { tip: NO_TIP });
+
+    await context.polkadotApi.tx.balances
+      .transfer(dorothy.address, GLMR)
+      .signAndSend(baltathar, { tip: MED_TIP });
+
+    await context.polkadotApi.tx.balances
+      .transfer(dorothy.address, GLMR)
+      .signAndSend(charleth, { tip: HIGH_TIP });
+
+    const result = await context.createBlock();
+    const hash = result.block.hash;
+    const apiAt = await context.polkadotApi.at(hash);
+    const { block } = await context.polkadotApi.rpc.chain.getBlock(hash);
+
+    // filter out inherent extrinsics, which should leave us with the ones we sent in their
+    // inclusion order
+    let transferExts = block.extrinsics.filter(
+      (ext) => ext.signer.toHex() !== "0x0000000000000000000000000000000000000000"
+    );
+
+    expect(transferExts.length).to.eq(3);
+    console.log(`transferEvents[0]: ${transferExts[0]}`);
+    expect(transferExts[0].tip.toBigInt()).to.eq(HIGH_TIP);
+    expect(transferExts[1].tip.toBigInt()).to.eq(MED_TIP);
+    expect(transferExts[2].tip.toBigInt()).to.eq(NO_TIP);
+  });
+
+  it("should treat eth and substrate txns fairly", async function () {
+    context.ethTransactionType = "EIP1559";
+
+    // tip 1 and 3 will be substrate txns, we express their tip above as per-gas but must send them
+    // expressed as a flat tip. So we query the weight and convert to gas, then multiply by the
+    // per-gas tip. We do this because it's the inverse of the txpool algo, and we want to show that
+    // this algo will order txns by tip in this test.
+    //
+    // so the expected order is:
+    // tip_0 (eth)
+    // tip_1 (substrate)
+    // tip_2 (eth)
+    // tip_3 (substrate)
+    const TIP_PER_GAS_0 = 10000n;
+    const TIP_PER_GAS_1 = 20000n;
+    const TIP_PER_GAS_2 = 30000n;
+    const TIP_PER_GAS_3 = 40000n;
+
+    // here we query the weight of a substrate balance transfer
+    const dummyTransfer = context.polkadotApi.tx.balances.transfer(alith.address, GLMR);
+    const info = await context.polkadotApi.rpc.payment.queryInfo(dummyTransfer.toHex());
+    const weight = info.weight.toBigInt();
+    const balances_transfer_effective_gas = weight / WEIGHT_PER_GAS;
+
+    // tx0 is an eth txn
+    const tx0 = await createTransfer(context, ethan.address, 1, {
+      from: charleth.address,
+      privateKey: CHARLETH_PRIVATE_KEY,
+      maxFeePerGas: HIGH_MAX_FEE_PER_GAS,
+      maxPriorityFeePerGas: "0x" + TIP_PER_GAS_0.toString(16),
```

Comment:
`nToHex(TIP_PER_GAS_0)`

Reply:
Ah, thanks, this is definitely the cumbersome way to do it.

I was bothered enough by it that I made `createTransaction` optionally take a `bigint` here: https://github.com/PureStake/moonbeam/pull/2066 -- I should take advantage of that in this PR :)